src/DynamoCore/Core/Threading/UpdateRenderPackageAsyncTask.cs

```diff
@@ -255,19 +259,27 @@ private void GetRenderPackagesFromMirrorData(MirrorData mirrorData, ref List<str
                     // color of 0,0,0,255 (Black), we adjust the color components here.
                     if (graphicItem is Curve || graphicItem is Surface || graphicItem is Solid || graphicItem is Point)
                     {
+                        var newLineColors = new List<byte>();
                         for (var i = 0; i < package.LineStripVertexColors.Count; i += 4)
                         {
```

Comment:
We don't directly tell the shader anything. We tell Helix and Helix tells the shader. We wrap all of our geometry in GeometryModel3D objects, which, for better or worse, expect vertex colors.

And, YES WE DO NEED IT. All points, lines, and meshes use vertex coloration blended with the phong color.